src/potentiometer-element.ts

```diff
@@ -0,0 +1,269 @@
+import { customElement, html, LitElement, property } from 'lit-element';
+
+interface Point {
+  x: number;
+  y: number;
+}
+
+let id = 1;
+
+/** The potentiometer SVG is taken from https://freesvg.org/potentiometer and some of the
+    functions are taken from https://github.com/vitaliy-bobrov/js-rocks knob component */
+@customElement('wokwi-potentiometer')
+export class PotentiometerElement extends LitElement {
+  @property() min = 0;
+  @property() max = 100;
+  @property() value = 0;
+  @property() step = 1;
+  @property() startDegree = -135;
+  @property() endDegree = 135;
+  private center: Point = { x: 0, y: 0 };
+  pressed = false;
+  id = `rotating-${id++}`;
+
+  clamp(min: number, max: number, value: number): number {
+    return Math.min(Math.max(value, min), max);
+  }
+
+  mapToMinMax(value: number, min: number, max: number): number {
+    return value * (max - min) + min;
+  }
+
+  percentFromMinMax(value: number, min: number, max: number): number {
+    return (value - min) / (max - min);
+  }
+
+  render() {
+    return html`
+      <style>
+        #${this.id} {
+          transform-origin: -120.168198px -140.5804px;
+          transform: rotate(var(--knob-angle, 0deg));
+        }
+      </style>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        width="20mm"
+        height="20mm"
+        viewBox="0 0 20 20"
+        @mousedown=${this.down}
+        @mousemove=${this.move}
+        @mouseup=${this.up}
+        @touchstart=${this.down}
+        @touchmove=${this.move}
+        @touchend=${this.up}
+      >
+        <g transform="translate(-6.7420057,-6.6693388)">
+          <g transform="matrix(0.15090139,0,0,0.15090139,34.862601,35.84314)">
+            <rect
+              ry="8.1522207"
+              y="-192.35448"
+              x="-185.37505"
+              height="128.89323"
+              width="129.11356"
+              style="opacity:1;fill:#045881;fill-opacity:1;stroke:#045881;stroke-width:1.95151615;stroke-linecap:butt;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
```

Comment:
Can you remove all the extraneous styles?  Stuff like `opacity:1` and `fill-opacity:1` is already the default. Basically, I think the only ones that you need to keep (without affecting the results) are:
- fill
- stroke
- stroke-width (only if stroke is not set to `none`). You can also round it (1.95 is enough, no need for all those extra decimal digits)
- opacity (only if < 1)
- stroke-opacity (only if < 1, stroke is not set to `none`)

This will clean up a lot of the clutter.

Reply:
On it